compiler/common-artifacts/CMakeLists.txt

```diff
@@ -0,0 +1,162 @@
+  set(RECIPE_SOURCE_PATH "${TFLITE_RECIPE_REPO}/${RECIPE}/${TEST_RECIPE_FILENAME}")
+  set(RECIPE_BINARY_PATH "${CMAKE_CURRENT_BINARY_DIR}/${RECIPE_FILE}")
+
+  set(TFLITE_FILE "${RECIPE}.tflite")
+  set(TFLITE_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/${TFLITE_FILE}")
+
+  # Copy .recipe
+  add_custom_command(OUTPUT ${RECIPE_BINARY_PATH}
+    COMMAND ${CMAKE_COMMAND} -E copy "${RECIPE_SOURCE_PATH}" "${RECIPE_BINARY_PATH}"
+    DEPENDS ${RECIPE_SOURCE_PATH}
+    COMMENT "Generate ${RECIPE_FILE}"
+  )
+
+  # Generate .tflite
+  add_custom_command(OUTPUT ${TFLITE_OUTPUT_PATH}
+    COMMAND $<TARGET_FILE:tflchef-file> ${RECIPE_BINARY_PATH} ${TFLITE_OUTPUT_PATH}
+    DEPENDS $<TARGET_FILE:tflchef-file> ${RECIPE_BINARY_PATH}
+    COMMENT "Generate ${TFLITE_FILE}"
+  )
+  set(MODEL_FORMAT "tflite")
+
+  list(APPEND TEST_DEPS ${RECIPE_BINARY_PATH} ${TFLITE_OUTPUT_PATH})
+
+  set(CIRCLE_FILE "${RECIPE}.circle")
+  set(CIRCLE_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/${CIRCLE_FILE}")
+
+  if(NOT DEFINED NO_CIRCLIZE_${RECIPE})
+    # Generate .circle
+    add_custom_command(OUTPUT ${CIRCLE_OUTPUT_PATH}
+      COMMAND $<TARGET_FILE:tflite2circle> ${TFLITE_OUTPUT_PATH} ${CIRCLE_OUTPUT_PATH}
+      DEPENDS $<TARGET_FILE:tflite2circle> ${TFLITE_OUTPUT_PATH}
+      COMMENT "Generate ${CIRCLE_FILE}"
+    )
+    set(MODEL_FORMAT "circle")
+    list(APPEND TEST_DEPS ${CIRCLE_OUTPUT_PATH})
+  endif()
+
+  set(OPT_CIRCLE_FILE "${RECIPE}.opt.circle")
+  set(OPT_CIRCLE_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/${OPT_CIRCLE_FILE}")
+
+  if(NOT DEFINED NO_OPTIMIZE_${RECIPE})
+    # Generate optimized .circle
+    add_custom_command(OUTPUT ${OPT_CIRCLE_OUTPUT_PATH}
+      COMMAND $<TARGET_FILE:circle2circle> --all ${CIRCLE_OUTPUT_PATH} ${OPT_CIRCLE_OUTPUT_PATH}
+      DEPENDS $<TARGET_FILE:circle2circle>  ${CIRCLE_OUTPUT_PATH}
+      COMMENT "Generate ${OPT_CIRCLE_FILE}"
+    )
+    set(OPT_FORMAT ".opt")
+    list(APPEND TEST_DEPS ${OPT_CIRCLE_OUTPUT_PATH})
+  endif()
+
+  set(MODEL_FILE "${RECIPE}${OPT_FORMAT}.${MODEL_FORMAT}")
+  set(MODEL_PATH "${CMAKE_CURRENT_BINARY_DIR}/${MODEL_FILE}")
+  set(NNPKG_FILE "${RECIPE}${OPT_FORMAT}")
+  set(NNPKG_PATH "${CMAKE_CURRENT_BINARY_DIR}/${NNPKG_FILE}")
+
+  add_custom_command(OUTPUT ${NNPKG_PATH}
+    COMMAND ${MODEL2NNPKG} ${MODEL_PATH}
+    DEPENDS ${MODEL2NNPKG} ${MODEL_PATH}
+    COMMENT "Generate ${RECIPE} nnpackage"
+  )
+  list(APPEND TEST_DEPS ${NNPKG_PATH})
+
+  set(INPUT_HDF5_FILE "${RECIPE}${OPT_FORMAT}.input.h5")
+  set(INPUT_BIN_PATH "${CMAKE_CURRENT_BINARY_DIR}/${INPUT_HDF5_FILE}")
+  
+  set(EXPECTED_HDF5_FILE "${RECIPE}${OPT_FORMAT}.expected.h5")
+  set(EXPECTED_BIN_PATH "${CMAKE_CURRENT_BINARY_DIR}/${EXPECTED_HDF5_FILE}")
+  
+  if(NOT DEFINED NO_TCGEN_${RECIPE})
+    # Generate input.h5, expected.h5
+    add_custom_command(OUTPUT ${INPUT_BIN_PATH} ${EXPECTED_BIN_PATH}
+      COMMAND $<TARGET_FILE:testDataGenerator> ${MODEL_FILE}
+      DEPENDS $<TARGET_FILE:testDataGenerator> ${MODEL_FILE}
```

Comment:
testDataGenerator should accept a circle file (am I right?). 

If a user does not generate a circle file using circlize(~), ${MODEL_FILE} will be a tflite model. (MODEL_FORMAT is tflite if NO_OPTIMIZE_${RECIPE} is defined)

Have you tested this case?

Reply:
@jinevening That's a good point! I didn't think of this.